src/prime_rl/trainer/weights.py

```diff
@@ -191,6 +238,11 @@ def save(
         """Save a HF-compatible weight-only checkpoint for a given step."""
         has_lora = has_lora_layers(model)
 
+        # Save LoRA adapters separately if configured
+        if self.config.save_adapter_separately and has_lora and self._is_master:
+            lora_state = self._get_adapter_state_dict(model)
+            self._save_lora_adapters(lora_state, model, step)
+
         cpu_state = self._gather_weights(model, dtype, has_lora_layers=has_lora)
```

Comment:
I think rn the gather still tries to merge even if we want to save the adapter separately. It shouldnt affect the result since the intended flow is that we only update adapters and therefore it doesnt matter what the cpu state is but might be good to not waste cycles merging

Reply:
We still merge ephemerally for full vllm shape compatibility due to limitations with vllm lora support